app/tools.py

```diff
@@ -0,0 +1,88 @@
+import torch
+import torch.nn as nn
+import fortepyan as ff
+from tqdm import tqdm
+from omegaconf import OmegaConf, DictConfig
+
+from model import make_model
+from data.quantizer import MidiQuantizer
+from utils import vocab_sizes, greedy_decode
+from data.dataset import quantized_piece_to_records
+from data.tokenizer import VelocityEncoder, QuantizedMidiEncoder
+
+
+def encode_record(sequence: dict, encoder: QuantizedMidiEncoder) -> torch.Tensor:
+    token_ids = encoder.encode(sequence)
+    token_ids = [encoder.token_to_id["<CLS>"]] + token_ids
+    return torch.tensor(token_ids, dtype=torch.int64)
+
```

Comment:
I think this should be a method of the encoder itself.

Reply:
The idea was to not append CLS token during encoding inside the encoder.

Reply:
My point is that from object oriented point of view, I'd prefer not to do:

```py
# No
def some_method(payload, some_object):
    return some_object.some_method(payload)
```

But go in this direction:

```py
# Yes
class SomeObject:
    def some_method(self, payload):
        ...
```

Also, since you are only calling this function in one place, you can also write the two lines of code explicitly there, and avoid obfuscation :)